### PR TITLE
add Microsoft.Test.UserTokenMock & tests

### PR DESCRIPTION
**Pull request by xieofxie**
## Description

Add user token mock for declarative test script.
It is required for testing bots that need authentication.

## Specific Changes

  - Add UserTokenMock class for token to mock
  - Add MockData as a helper for mocks that want to be loaded from file and don't have variations. (subtle as it conflicts with schema)
  - Add `List<MockData<UserTokenMock>>` in TestScript

## Testing

  - Add TestScriptTests_UserTokenMock.test.dialog

Thanks!

### CODE REVIEW

**Comment by xieofxie** [comment]
I add Microsoft.Test.UserTokenMock for Microsoft.OAuthInput.
Also I increase time interval in ShowTyping_TestMiddleware_1_Second_Interval for unlucky ones..
Please review this, thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/TestScriptTests/TestScriptTests_UserTokenMock.test.dialog [hunk 1]
```diff
@@ -0,0 +1,106 @@
+﻿{
+    "$schema": "../../../tests.schema",
+    "$kind": "Microsoft.Test.Script",
+    "description": "Test UserTokenMock",
+    "userTokenMocks": [
+        "magiccode.mock",
+        "nomagiccode.mock"
+    ],
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/TestScriptTests/TestScriptTests_UserTokenMock.test.dialog:8]: this is pretty cool, thanks for adding this!

**Comment by xieofxie** [comment]
@tomlm Thanks for reviewing this! But forgive me, I regretted about using MockData as it made TestScript class ugly. So I update the code to use abstract class UserTokenMock and now the implementation is Microsoft.Test.UserTokenBasicMock.

Also since we revert QnAMaker here 266a746e28b83ee3a8603e9d4be3f3163225b8b2, properties are removed in tests.schema.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenBasicMock.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using Microsoft.Bot.Builder.Adapters;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.UserTokenMocks
+{
+    public class UserTokenBasicMock : UserTokenMock
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenBasicMock.cs:10]: Public classes need /// comments

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenMock.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Builder.Dialogs.Debugging;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.UserTokenMocks
+{
+    public abstract class UserTokenMock
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenMock.cs:9]: needs /// comments

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenBasicMock.cs (hunk 2 above)
Comment (xieofxie) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/UserTokenMocks/UserTokenBasicMock.cs:10]: Resolved here https://github.com/microsoft/botbuilder-dotnet/pull/4226

**Pull request merged**